core/src/main/scala/org/apache/spark/eventhubs/common/client/EventHubsClientWrapper.scala

```diff
@@ -18,69 +18,61 @@
 package org.apache.spark.eventhubs.common.client
 
 import java.net.URI
-import java.time.{ Duration, Instant }
+import java.time.Instant
+import java.util.concurrent.ConcurrentHashMap
 
 import scala.collection.JavaConverters._
-import EventHubsOffsetTypes.EventHubsOffsetType
 import com.microsoft.azure.eventhubs._
-import org.apache.spark.eventhubs.common.NameAndPartition
+import org.apache.spark.eventhubs.common.EventHubsConf
 import org.apache.spark.internal.Logging
 
-import scala.util.{ Failure, Success, Try }
+import scala.util.Try
 
 /**
  * Wraps a raw EventHubReceiver to make it easier for unit tests
  */
 @SerialVersionUID(1L)
-private[spark] class EventHubsClientWrapper(private val ehParams: Map[String, String])
+private[spark] class EventHubsClientWrapper(private val ehConf: EventHubsConf)
     extends Serializable
     with Client
     with Logging {
 
-  override private[spark] var client: EventHubClient = _
-  private[spark] var partitionReceiver: PartitionReceiver = _
+  import org.apache.spark.eventhubs.common._
+
+  // Extract relevant info from ehParams
+  private val ehNamespace = ehConf.namespace.get
+  private val ehName = ehConf.name.get
+  private val ehPolicyName = ehConf.keyName.get
+  private val ehPolicy = ehConf.key.get
+  private val consumerGroup = ehConf.consumerGroup.getOrElse(DefaultConsumerGroup)
 
-  /* Extract relevant info from ehParams */
-  private val ehNamespace = ehParams("eventhubs.namespace").toString
-  private val ehName = ehParams("eventhubs.name").toString
-  private val ehPolicyName = ehParams("eventhubs.policyname").toString
-  private val ehPolicy = ehParams("eventhubs.policykey").toString
-  private val consumerGroup =
-    ehParams.getOrElse("eventhubs.consumergroup", EventHubClient.DEFAULT_CONSUMER_GROUP_NAME)
   private val connectionString =
     Try {
-      new ConnectionStringBuilder(ehNamespace, ehName, ehPolicyName, ehPolicy).toString
+      new ConnectionStringBuilder(ehNamespace, ehName, ehPolicyName, ehPolicy)
     } getOrElse Try {
-      new ConnectionStringBuilder(new URI(ehNamespace), ehName, ehPolicyName, ehPolicy).toString
+      new ConnectionStringBuilder(new URI(ehNamespace), ehName, ehPolicyName, ehPolicy)
```

Comment:
I was thinking about this earlier and would it make more sense to have an optional "suffix" config parameter? Rather than doubling execution and trying to parse the full namespace we could have a suffix that defaults to the usual EH value if not present.

Reply:
I like this idea! I'll get to it after tests are finished.